### PR TITLE
Sass Formatter

### PR DESCRIPTION
**Pull request by TheRealSyler**
Related to #1433

not sure if the return values of the unused functions in the `SassLanguageMode` are an issue.

### CODE REVIEW

File: server/src/modes/style/sass/sassLanguageMode.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  }
+  format(document: TextDocument, range: Range, formattingOptions: FormattingOptions) {
+    let sassConfig: any = {
+      convert: true,
+      deleteCompact: true,
+      deleteEmptyRows: true,
+      deleteWhitespace: true,
+      replaceSpacesOrTabs: true,
+      setPropertySpace: true
+    };
```
Comment (octref) [inline server/src/modes/style/sass/sassLanguageMode.ts]: Can you extract this into an interface exported from this file? We might have a default config at some point.

File: server/src/modes/style/sass/sassLanguageMode.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if (this.config.sass && this.config.sass.format) {
+      sassConfig = this.config.sass.format;
```
Comment (octref) [inline server/src/modes/style/sass/sassLanguageMode.ts]: Use `Object.assign(this.config.sass.format, sassConfig)` to merge them.

**Comment by octref** [comment]
Good work. Thanks a lot!

I have added docs and pushed an integration test. Left some comments as well.

File: server/src/modes/style/sass/sassLanguageMode.ts (hunk 1 above)
Comment (TheRealSyler) [reply server/src/modes/style/sass/sassLanguageMode.ts]: i already made the interface [here](https://github.com/TheRealSyler/sass-formatter/blob/master/src/format.provider.ts#L32), i can just import the it from `sass-formatter`, maintaining two instances of the same interface seems unnecessary.

**Comment by TheRealSyler** [comment]
i couldn't test it locally because i installed linux, and i get an error when i try to start the debug server, but this should address your comments, and in theory it should work without braking anything, but every time i think this should work something brakes.

File: server/src/modes/style/sass/sassLanguageMode.ts (hunk 2 above)
Comment (TheRealSyler) [reply server/src/modes/style/sass/sassLanguageMode.ts]: added in the newest commit.

**Comment by octref** [comment]
Retriggering CI

**Comment by TheRealSyler** [comment]
moved to #1658

**Pull request closed without merging**